xconnect_backend/library/views.py

```diff
@@ -0,0 +1,36 @@
+from django.shortcuts import render
+# from django.views.generic import ListView
+from rest_framework import generics,permissions
+from .models import Books,Reservation
+from .serializer import BookSerializer,ReservationViewSerializer,ReservationSerializer
+from .permissions import is_libraryStaff
+
+#for html templet view
+# class BooksListView(ListView):
+#     model = Books
+#     template_name = "Books_list.html"
+
+class BooksListView(generics.ListCreateAPIView):
+    permission_classes = [is_libraryStaff]
+    queryset = Books.objects.all()
+    serializer_class = BookSerializer
+
+class BookListView(generics.RetrieveUpdateDestroyAPIView):
+    queryset = Books.objects.all()
+    serializer_class = BookSerializer
+
+class  ReservationListView(generics.ListCreateAPIView):
+    queryset = Reservation.objects.all()
```

Comment:
add permission if user is authenticated also add permission so only the user that has reserved can see what they reserved and other user can't see what he/she has reserved